From: Roland Dreier <roland@purestorage.com>
Date: Wed, 29 Feb 2012 20:40:46 +0000
Subject: [PATCH 38/58] be2net: Remove unused OFFSET_IN_PAGE() macro

commit 97767a87f3be8834192dc3fc9412aaccf708d87f upstream.

Signed-off-by: Roland Dreier <roland@purestorage.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
---
 drivers/net/ethernet/emulex/benet/be.h |    4 ----
 1 file changed, 4 deletions(-)

Index: linux-3.2.46/drivers/net/ethernet/emulex/benet/be.h
===================================================================
--- linux-3.2.46.orig/drivers/net/ethernet/emulex/benet/be.h	2013-07-26 19:21:41.000000000 +0000
+++ linux-3.2.46/drivers/net/ethernet/emulex/benet/be.h	2013-07-26 19:21:42.000000000 +0000
@@ -442,10 +442,6 @@
 		((u32)((((size_t)(_address) & (PAGE_SIZE_4K - 1)) + 	\
 			(size) + (PAGE_SIZE_4K - 1)) >> PAGE_SHIFT_4K))
 
-/* Byte offset into the page corresponding to given address */
-#define OFFSET_IN_PAGE(addr)						\
-		 ((size_t)(addr) & (PAGE_SIZE_4K-1))
-
 /* Returns bit offset within a DWORD of a bitfield */
 #define AMAP_BIT_OFFSET(_struct, field)  				\
 		(((size_t)&(((_struct *)0)->field))%32)
