commit f9b5246fd33f96fe2a31dbd74dce8d6da7df0fbc
Author: popcornmix <popcornmix@gmail.com>
Date:   Wed Jul 11 13:55:11 2012 +0100

    Add missing UART0_CLOCK from last commit

Index: linux-3.2.46/arch/arm/mach-bcm2708/bcm2708.c
===================================================================
--- linux-3.2.46.orig/arch/arm/mach-bcm2708/bcm2708.c	2013-07-26 19:33:54.000000000 +0000
+++ linux-3.2.46/arch/arm/mach-bcm2708/bcm2708.c	2013-07-26 19:34:01.000000000 +0000
@@ -183,7 +183,7 @@
  * These are fixed clocks.
  */
 static struct clk ref24_clk = {
-	.rate = 3000000,	/* The UART is clocked at 3MHz via APB_CLK */
+	.rate = UART0_CLOCK,	/* The UART is clocked at 3MHz via APB_CLK */
 };
 
 static struct clk osc_clk = {
Index: linux-3.2.46/arch/arm/mach-bcm2708/include/mach/platform.h
===================================================================
--- linux-3.2.46.orig/arch/arm/mach-bcm2708/include/mach/platform.h	2013-07-26 19:33:54.000000000 +0000
+++ linux-3.2.46/arch/arm/mach-bcm2708/include/mach/platform.h	2013-07-26 19:34:01.000000000 +0000
@@ -216,7 +216,7 @@
 #define PM_RSTS_HADDRF_SET                                 0x00000002
 #define PM_RSTS_HADDRQ_SET                                 0x00000001
 
-
+#define UART0_CLOCK      3000000
 
 #endif
 
