commit 2b4c39f7d2bb2d0b47ddd4c40250d54f1f3551aa
Author: Dom Cobley <dc4@broadcom.com>
Date:   Mon Mar 26 22:18:37 2012 +0100

    Cast to avoid warning

Index: linux-3.2.46/drivers/video/bcm2708_fb.c
===================================================================
--- linux-3.2.46.orig/drivers/video/bcm2708_fb.c	2013-07-26 19:32:21.000000000 +0000
+++ linux-3.2.46/drivers/video/bcm2708_fb.c	2013-07-26 19:32:32.000000000 +0000
@@ -191,7 +191,7 @@
 
 static int bcm2708_fb_set_par(struct fb_info *info)
 {
-	unsigned val = 0;
+	uint32_t val = 0;
 	struct bcm2708_fb *fb = to_bcm2708(info);
 	volatile struct fbinfo_s *fbinfo = fb->info;
 	fbinfo->xres = info->var.xres;
@@ -243,7 +243,7 @@
 	    ("BCM2708FB: start = %p,%p width=%d, height=%d, bpp=%d, pitch=%d size=%d success=%d\n",
 	     (void *)fb->fb.screen_base, (void *)fb->fb.fix.smem_start,
 	     fbinfo->xres, fbinfo->yres, fbinfo->bpp,
-	     fbinfo->pitch, fb->fb.screen_size, val);
+	     fbinfo->pitch, (int)fb->fb.screen_size, val);
 
 	return val;
 }
