Index: linux-3.2.46/security/device_cgroup.c
===================================================================
--- linux-3.2.46.orig/security/device_cgroup.c	2013-07-26 19:16:53.000000000 +0000
+++ linux-3.2.46/security/device_cgroup.c	2013-07-26 19:17:04.000000000 +0000
@@ -14,6 +14,7 @@
 #include <linux/slab.h>
 #include <linux/rcupdate.h>
 #include <linux/mutex.h>
+#include <linux/export.h>
 
 #define ACC_MKNOD 1
 #define ACC_READ  2
