From bfd1622f13d5b709d3b0863de3b065232f56aa69 Mon Sep 17 00:00:00 2001
From: Clark Williams <clark.williams@gmail.com>
Date: Fri, 8 Dec 2023 10:49:54 -0600
Subject: [PATCH 62/62] Linux 6.1.66-rt19 REBASE
Origin: https://www.kernel.org/pub/linux/kernel/projects/rt/6.1/older/patches-6.1.66-rt19.tar.xz

Signed-off-by: Clark Williams <clark.williams@gmail.com>
---
 localversion-rt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/localversion-rt b/localversion-rt
index 9f7d0bdbffb1..483ad771f201 100644
--- a/localversion-rt
+++ b/localversion-rt
@@ -1 +1 @@
--rt13
+-rt19
-- 
2.43.0

