From: "Luis Claudio R. Goncalves" <lgoncalv@redhat.com>
Date: Sun, 15 Sep 2024 21:42:37 -0300
Subject: [PATCH 323/323] Linux 5.10.225-rt117 REBASE
Origin: https://git.kernel.org/cgit/linux/kernel/git/rt/linux-stable-rt.git/commit?id=bcbe2ad91cc44f51fa0c8a35147306b1eb11a2db

Signed-off-by: Luis Claudio R. Goncalves <lgoncalv@redhat.com>
---
 localversion-rt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/localversion-rt b/localversion-rt
index 21988f9ad53f..9788245dd428 100644
--- a/localversion-rt
+++ b/localversion-rt
@@ -1 +1 @@
--rt34
+-rt117
