From: "Luis Claudio R. Goncalves" <lgoncalv@redhat.com>
Date: Mon, 22 Jun 2026 21:23:29 -0300
Subject: [PATCH 328/328] Linux 5.10.259-rt155 REBASE
Origin: https://git.kernel.org/cgit/linux/kernel/git/rt/linux-stable-rt.git/commit?id=3cf8563c72bc868ffb1273545882a56d31805d59

Signed-off-by: Luis Claudio R. Goncalves <lgoncalv@redhat.com>
---
 localversion-rt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/localversion-rt b/localversion-rt
index 21988f9ad53f..a0eb4ea28f15 100644
--- a/localversion-rt
+++ b/localversion-rt
@@ -1 +1 @@
--rt34
+-rt155
