From: Clark Williams <clrkwllms@kernel.org>
Date: Tue, 23 Jun 2026 21:51:50 -0500
Subject: [PATCH 72/72] Linux 6.1.176-rt64 REBASE
Origin: https://git.kernel.org/cgit/linux/kernel/git/rt/linux-stable-rt.git/commit?id=0361d6d83b18115cf7650effbbf9abbe7a55b245

Signed-off-by: Clark Williams <clrkwllms@kernel.org>
---
 localversion-rt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/localversion-rt b/localversion-rt
index 40d81d8e61b6..10474042df49 100644
--- a/localversion-rt
+++ b/localversion-rt
@@ -1 +1 @@
--rt62
+-rt64
