From: "Luis Claudio R. Goncalves" <lgoncalv@redhat.com>
Date: Thu, 26 Feb 2026 09:04:35 -0300
Subject: [PATCH 327/327] Linux 5.10.251-rt146 REBASE
Origin: https://git.kernel.org/cgit/linux/kernel/git/rt/linux-stable-rt.git/commit?id=7b92110e913b9e752a14e7e763dbfb1635408b3d

Signed-off-by: Luis Claudio R. Goncalves <lgoncalv@redhat.com>
---
 localversion-rt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/localversion-rt b/localversion-rt
index 21988f9ad53f..30edde2fc1d2 100644
--- a/localversion-rt
+++ b/localversion-rt
@@ -1 +1 @@
--rt34
+-rt146
